app/core/src/main/java/com/fsck/k9/preferences/ServerSettingsDescriptions.kt

```diff
@@ -0,0 +1,112 @@
+package com.fsck.k9.preferences
+
+import com.fsck.k9.preferences.Settings.IntegerRangeSetting
+import com.fsck.k9.preferences.Settings.SettingsDescription
+import com.fsck.k9.preferences.Settings.SettingsUpgrader
+import com.fsck.k9.preferences.Settings.StringSetting
+import com.fsck.k9.preferences.upgrader.ServerSettingsUpgraderTo92
+import java.util.TreeMap
+
+/**
+ * Contains information to validate imported server settings with a given content version, and to upgrade those server
+ * settings to the latest content version.
+ */
+@Suppress("MagicNumber")
+internal class ServerSettingsDescriptions {
```

Comment:
The name was chosen mostly for consistency. We already have `GeneralSettingsDescriptions`, `AccountSettingsDescriptions`, `FolderSettingsDescriptions`, and `IdentitySettingsDescriptions`.

These classes are using the plural form `Descriptions` because they are collections of `SettingsDescription` instances (which probably should use `Setting` instead of the plural form :+1:).

Do you have a suggestion on what to do with these classes? We could try to find a more fitting name or maybe move the list of "setting descriptions" and the list of "upgraders" into different classes.

Reply:
Finding a name that fits all it's current use cases is hard, also a sign that it does to much. Splitting would be the best option.